backtesting/backtesting.py

```diff
@@ -493,7 +503,8 @@ def _open_position(self, price, is_long):
 
         i, price = self._get_market_price(price)
 
-        position = float(self._cash * self._leverage / (price * (1 + self._commission)))
+        size = self._cash if self.orders._size is None else self.orders._size
+        position = float(size * self._leverage / (price * (1 + self._commission)))
```

Comment:
Is this `position` computation correct? You effectively replace former `cash` with `size`, but size actually corresponds to `cash / price`?

I think when size is given, position might better be `float(size) if enough_cash else error`?

Reply:
The intention for the `size` variable being computed as such is because:

1. If size is not specified when performing an order, it will default to the default behaviour of using entire holding cash amount.
2. I did not notice there was a `margin` option to define a leveraged trade. (Only found out about it after reading: #10 ) Was initially under the assumption that all trades were on a 1:1 leverage. Under my assumption, i believed that specifying the `margin` option wasn't needed in order to define a "leveraged trade". In fact, by specifying an order size that is larger than your holding cash amount would technically mean you enter a trade on cross margin. (which was why I did not implement a check for `if enough_cash`). Reason for this assumption is largely influenced by Trading View's pine script, and how "leveraged trades" would be entered under their backtesting tools.

So my thoughts are as follows, and would need some clarification:

1. Does the library actually calculate liquidations. If it does, the `margin` option would come in handy for that. A quick look into the code base shows how `margin` is used to calculate `leverage` amount. But `margin` and `leverage` are simply being used in reporting functions and likely no where else?
2. Would it make sense to have margin be calculated dynamically if order size was specified, and larger than than holding cash. In that case, we could dynamically calculate a cross margin on init instead.

Would love to take a stab at pyramiding too, there are indeed many use cases for that. Let me know your thoughts. I might be working on a path different than the philosophy of this library. :)

Reply:
> Does the library actually calculate liquidations.

It simulates them.

https://github.com/kernc/backtesting.py/blob/b3cfaff2956d19b3c4ea6329bb915de205fd7032/backtesting/backtesting.py#L561-L570

`margin=` therefore represents initial and maintenance margin of the trades. It's used only in stats and plots so the interested user gets a rough idea on how leveraged their strategy is allowed to be to still remain afloat.

Reply:
In our case, `position` is actually equal to `size` when the user requests it. Thus, we only need to deduct the commission and then confirm we have (leveraged) cash to cover it all.

Reply:
I'm really sorry for the long time I took to respond. I'm completely elsewhere, and this seemed like a much harder issue to grasp than it really was.

Reply:
What do you think? :sweat_smile: